mysql-test/t/audit_log_filter_users.test

```diff
@@ -0,0 +1,83 @@
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_exclude_accounts= '22useer@localhost';
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_include_accounts= 'user1@localhost, user2@localhost, user3@localhost';
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_include_accounts= '';
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_include_accounts= NULL;
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+
+SET GLOBAL audit_log_exclude_accounts= "'us,er1'@'localhost',, user22@127.0.0.1,admin@%";
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_include_accounts= '22useer@localhost';
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_exclude_accounts= 'user1@localhost, user2@localhost, user3@localhost';
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_exclude_accounts= '';
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+SET GLOBAL audit_log_exclude_accounts= NULL;
+SELECT @@audit_log_include_accounts, @@audit_log_exclude_accounts;
+
+let $MYSQLD_DATADIR= `select @@datadir`;
+let MYSQLD_DATADIR= $MYSQLD_DATADIR;
+
+SET GLOBAL audit_log_include_accounts= ',,';
+connect (silent,localhost,root,,);
+connection default;
+
+SET GLOBAL audit_log_flush=ON;
+--remove_file $MYSQLD_DATADIR/test_audit.log
+SET GLOBAL audit_log_flush=ON;
+
+connection silent;
+SET GLOBAL audit_log_include_accounts= 'user1@localhost,, user22@127.0.0.1,admin@%';
+connect (test,localhost,root,,);
+connection test;
+SELECT 'skip';
+disconnect test;
+connection silent;
+
+SET GLOBAL audit_log_include_accounts= 'user1@localhost,root@localhost , user22,admin';
+connect (test,localhost,root,,);
+connection test;
+SELECT 'pass';
+disconnect test;
+connection silent;
+
+SET GLOBAL audit_log_exclude_accounts= 'user1@localhost,, user22@127.0.0.1,admin@%';
+connect (test,localhost,root,,);
+connection test;
+SELECT 'pass';
+disconnect test;
+connection silent;
+
+SET GLOBAL audit_log_exclude_accounts= 'user1@localhost, \'root\'@localhost, user22,admin';
+connect (test,localhost,root,,);
+connection test;
+SELECT 'skip';
+disconnect test;
+disconnect silent;
+connection default;
+
+SET GLOBAL audit_log_exclude_accounts= NULL;
+
+--move_file $MYSQLD_DATADIR/test_audit.log $MYSQLD_DATADIR/test_audit_csv.log
+set global audit_log_flush= ON;
+perl;
+  print "================================\n";
+  open my $file, $ENV{'MYSQLD_DATADIR'} . '/test_audit_csv.log' or die "Could not open log: $!";
+  while ($line = <$file>) {
+    if ($line =~ /",0,"([^"]+)"/) {
+      print "$1\n";
+    }
+  };
+  close $file;
+  print "================================\n";
+EOF
+--remove_file $MYSQLD_DATADIR/test_audit.log
+--remove_file $MYSQLD_DATADIR/test_audit_csv.log
```

Comment:
The testcase only ever connects with "root",  thus it cannot possibly cover the functionality, please create some users, and merge the "parser" (lines 4-24) tests with functionality tests, using different users